### PR TITLE
Add support to output magnets to stdout

### LINKED ISSUE #54: Add support for local torrent client or exporting magnet links
Would it be possible to add an option to print to stdout the magnet links so they could be added to the local torrent client?

### PR DESCRIPTION
**Pull request by aTosser**
:)

### CODE REVIEW

File: HorribleDownloader/cmd.py [hunk 1]
```diff
@@ -224,12 +224,15 @@ def should_download(episode):
                 if inp not in ('', 'Y', 'y', 'yes', 'Yes'):
                     print(fg(1) + 'aborting download\n' + fg.rs)
                     exit(1)
-
-        for episode in filtered_episodes:
-            download(episode, qualities, config.download_dir)
+                          
+        if args.export:
```
Comment (mtshrmn) [inline HorribleDownloader/cmd.py:232]: You must first add an argument for export, it must have an understandable description

File: HorribleDownloader/cmd.py [hunk 2]
```diff
@@ -224,12 +224,15 @@ def should_download(episode):
...
+            for episode in filtered_episodes:
+                print(episode[quality]['Magnet'])
```
Comment (mtshrmn) [inline HorribleDownloader/cmd.py]: There can be multiple qualities, they live in the `qualities` variable
```py
for episode in filtered_episodes:
   for quality in qualities:
      print(episode[quality]['Magnet'])
```

File: HorribleDownloader/cmd.py [hunk 3]
```diff
@@ -280,12 +283,16 @@ def should_download(episode):
             logger.info("user has aboorted the download")
             exit(1)
 
-
-    for episode in downloads_list:
-        download(episode, qualities, config.download_dir)
-        config.update_entry(episode["title"], episode["episode"])
-        logger.info(f'updated entry: {episode["title"]} - {episode["episode"]}')
+    if args.export:
+        for episode in downloads_list:
```
Comment (mtshrmn) [inline HorribleDownloader/cmd.py:292]: Same goes here, you must iterate over the qualities.

Note: update the entry once all qualities were printed out.

**Review by mtshrmn — CHANGES_REQUESTED** [review]

**Comment by mtshrmn** [comment]
Before merging this, please update the README to document this feature.

File: HorribleDownloader/cmd.py (hunk 1 above)
Comment (aTosser) [reply HorribleDownloader/cmd.py:232]: My version has this, I was quick and did the edit manually again here on the web interface and missed that line, I will correct.

**Comment by aTosser** [comment]
Please review. The style matching changes should solve the config entry updating after all of the qualities are printed.

If you would like I can add an "are you sure" to the entry update (obv with the skip checks) so it can be confirmed by the user prior to updating the entry. This would allow for a re-run if something goes awry.

**Comment by mtshrmn** [comment]
I have totally forgot about it in the last review. If you want to print out the magnets, you must suppress all other prints. 
For [example](https://github.com/mtshrmn/horrible-downloader/blob/81fa030e8bd852884116fd483e1615d5dc46b574/HorribleDownloader/cmd.py#L205)
```py
# after modification
if not args.quiet and not args.export:
    print(f"{fg(3)}FETCHING:{fg.rs} {title}")
```

**Comment by aTosser** [comment]
Not necesarily, in my current use case using the -d flag in conjunction it is kind of useful to have the display of episodes found at the end of the search phase. (to guarantee that the anime is the correct one.) But, I am manually copying the magnets to the client. The reason I thought of silencing the output was for scripted usage to get clean standard output.

I'll add some checks for quiet so scripting won't be affected.

**Comment by mtshrmn** [comment]
I suppose you can use the `--quiet` flag with `--export` to achieve the desired effect. Please document that behavior in the README as well.

**Comment by aTosser** [comment]
Readme updated, I do not think an aditional print to ask whether to update the config is necessary as the output from first run will still be present in the terminal. This is only useful to someone in my usecase who want to manually copy the magnets. For scripting I feel it should happen automatically and if it is desired to preserve the config there should instead be a --dryrun option added if it is required.

**Comment by mtshrmn** [comment]
Thanks for contributing 😄

**Pull request merged**